File: Section Reports/Layout/SubReport/C#/SubReport/rptBookmarkMain.cs
using System;
namespace GrapeCity.ActiveReports.Samples.SubReport
{
	/// <summary>
	/// rptBookmarkMain description overview
	/// 
	/// </summary>
	public partial class rptBookmarkMain : GrapeCity.ActiveReports.SectionReport
	{
		rptBookmarkSub _rpt;
		string _categoryID;
		public rptBookmarkMain()
		{
			//
			// ActiveReport designer support is required.
			//
			//
			InitializeComponent();
		}
		private void rptBookmarkMain_FetchData(object sender, FetchEventArgs eArgs)
		{
			_categoryID = Fields["CategoryID"].Value.ToString();
		}
		private void Detail_Format(object sender, EventArgs e)
		{
			Data.OleDBDataSource subDS = new Data.OleDBDataSource();
			subDS.ConnectionString = ((Data.OleDBDataSource)DataSource).ConnectionString;
			subDS.SQL = "Select * from products where categoryID = " + _categoryID;
			_rpt.DataSource = subDS;
			ctlSubreport.Report = _rpt;
			Detail.AddBookmark(txtCategoryName.Text);
		}
		private void rptBookmarkMain_ReportStart(object sender, EventArgs e)
		{
			_rpt = new rptBookmarkSub();
			Data.OleDBDataSource dS = new Data.OleDBDataSource();
			dS.ConnectionString = Properties.Resources.ConnectionString;
			dS.SQL = "Select * from categories";
			DataSource = dS;
			
			reportInfo1.FormatString = "Page {PageNumber} of {PageCount}";
			
			
		}
	}
}

File: Section Reports/Layout/SubReport/VB.NET/SubReport/rptBookmarkMain.vb
Public Class rptBookmarkMain
	Inherits GrapeCity.ActiveReports.SectionReport
	Dim rpt As rptBookmarkSub
	Dim _categoryID As String
	Public Sub New()
		MyBase.New()
		InitializeComponent()
	End Sub
	Private Sub rptMainBM_FetchData(ByVal sender As Object, ByVal eArgs As FetchEventArgs) Handles MyBase.FetchData
		_categoryID = (Fields("CategoryID").Value).ToString()
	End Sub
	Private Sub Detail_Format(ByVal sender As Object, ByVal e As EventArgs) Handles Detail.Format
		Dim subDS As New Data.OleDBDataSource()
		subDS.ConnectionString = CType(DataSource, Data.OleDBDataSource).ConnectionString
		subDS.SQL = "Select * from products where categoryID = " + _categoryID
		rpt.DataSource = subDS
		ctlSubreport.Report = rpt
		Detail.AddBookmark(txtCategoryName.Text)
	End Sub
	
	Private Sub rptBookmarkMain_ReportStart(ByVal sender As System.Object, ByVal e As EventArgs) Handles MyBase.ReportStart
		rpt = New rptBookmarkSub()
		Dim dS As New Data.OleDBDataSource()
		dS.ConnectionString = My.Resources.ConnectionString
		dS.SQL = "Select * from categories"
		DataSource = dS
		
		ReportInfo1.FormatString = "Page {PageNumber} of {PageCount}"
		
		'
	End Sub
End Class
